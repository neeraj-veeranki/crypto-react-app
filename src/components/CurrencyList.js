import React from 'react';
import { makeStyles, withStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';

const useStyles = makeStyles({
  table: {
    minWidth: 350,
  },
});

const StyledTableCell = withStyles((theme) => ({
    head: {
      backgroundColor: '#3f50b5',
      color: theme.palette.common.white,
    },
    body: {
      fontSize: 14,
    },
  }))(TableCell);
  
  const StyledTableRow = withStyles((theme) => ({
    root: {
      '&:nth-of-type(odd)': {
        backgroundColor: theme.palette.action.hover,
      },
    },
  }))(TableRow);

export default function CurrencyList(props) {
  const {currencyData} = props;
  const classes = useStyles();

  return (
    <TableContainer component={Paper}>
      <Table className={classes.table} aria-label="simple table">
        <TableHead>
          <TableRow>
            <StyledTableCell>CurrencyCode</StyledTableCell>
            <StyledTableCell align="right">Quotes</StyledTableCell>
          </TableRow>
        </TableHead>
        <TableBody>
        {currencyData.map((row) => (
            <StyledTableRow key={row.currency}>
              <StyledTableCell component="th" scope="row">
                {row.currency}
              </StyledTableCell>
              <TableCell align="right">{row.quote}</TableCell>
            </StyledTableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}
