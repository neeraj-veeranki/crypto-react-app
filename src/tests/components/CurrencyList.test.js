import React from "react";
import CurrencyList from "../../components/CurrencyList";
import { render, cleanup } from '@testing-library/react';
afterEach(cleanup)
test('shows data received from props', async() => {
    const props = {
        currencyData: [{
            'currency': 'USD',
            'quote': '123'
        }]
    }
  const { getByText } = render(<CurrencyList {...props}/>)
  expect(getByText(/CurrencyCode/i)).toBeVisible();;
});
