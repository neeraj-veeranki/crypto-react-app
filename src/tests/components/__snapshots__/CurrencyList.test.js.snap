// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyList after passing prop data table should have necessary rows 1`] = `"<div class=\\"MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded\\"><table class=\\"MuiTable-root makeStyles-table-1\\" aria-label=\\"simple table\\"><thead class=\\"MuiTableHead-root\\"><tr class=\\"MuiTableRow-root MuiTableRow-head\\"><th class=\\"MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-2\\" scope=\\"col\\">CurrencyCode</th><th class=\\"MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-2 MuiTableCell-alignRight\\" scope=\\"col\\">Quotes</th></tr></thead><tbody class=\\"MuiTableBody-root\\"><tr class=\\"MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-4\\"><th class=\\"MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3\\" role=\\"cell\\" scope=\\"row\\">USD</th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight\\">123</td></tr></tbody></table></div>"`;
